lib/clone.js

```diff
@@ -49,21 +52,24 @@ const cloneTask = (sshGitHubUrl, repoName, progressBar) => {
 const clone = (services, task = cloneTask) => {
   process.stdout.write(`Cloning repos...${newline}`)
 
-  const gitHubDetails = getGitHubDetails(services)
+  const gitHubRepoInfo = getGitHubDetails(services)
   const progressBar = new Progress(':percent [:bar] (:current/:total)', {
-    total: gitHubDetails.length,
+    total: gitHubRepoInfo.length,
     clear: true
   })
+  const githubInfoParentArray = chunkArray(gitHubRepoInfo, numberOfCPUs * 4)
```

Comment:
> Yeah it feels like this might be unnecessarily punishing

agreed - thats why I did `* 4` to keep it closer to the original value of `20` but with a variable so as to help out machines with less cpu's.

>I didn't look to much into it yet, but something like tracking the memory needed by each clone, averaging it, checking against freemem, then polling it for when there's enough to kick off another.

Sounds interesting. Something for another PR? fancy raising an issue against this when its in `master`?